File: app/src/main/java/moe/yuuta/mipushtester/push/APIInterface.java
package moe.yuuta.mipushtester.push;

import com.google.gson.JsonObject;

import moe.yuuta.mipushtester.update.Update;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;

public interface APIInterface {
    @POST("/test")
    Call<JsonObject> push (@Body PushRequest request);

    @GET("/update")
    Call<Update> getUpdate ();
}

File: app/src/main/kotlin/api/APIInterface.kt
package moe.yuuta.mipushtester.api

import com.google.gson.JsonObject
import moe.yuuta.mipushtester.push.PushRequest
import moe.yuuta.mipushtester.topic.Topic
import moe.yuuta.mipushtester.update.Update
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST

interface APIInterface {
    @POST("/test")
    fun push(@Body request: PushRequest): Call<JsonObject>

    @GET("/update")
    fun getUpdate(): Call<Update>

    @GET("/test/topic")
    fun getAvailableTopics(): Call<MutableList<Topic>>
}